pmlc/rt/jit_executable.cc

```diff
@@ -386,10 +389,18 @@ class JitExecutable final : public Executable {
 
     std::copy(preParams.begin(), preParams.end(), std::back_inserter(ptrs));
 
-    descriptors.reserve(bufptrs.size());
-    for (unsigned i = 0; i < bufptrs.size(); i++) {
-      descriptors.emplace_back(bufptrs[i], program->arguments[i].shape);
-      ptrs.push_back(descriptors[i].ptr());
+    for (auto [type, buffer] : llvm::zip(program->inputs, inputBuffers)) {
+      descriptors.emplace_back(buffer->data(), type.cast<RankedTensorType>());
+      ptrs.push_back(descriptors.back().ptr());
+    }
+    for (const compiler::ConstantArgument &arg : program->constants) {
```

Comment:
@earhart Hopefully this answers your question on how constant buffers are currently handled. They come from the `Program` and are injected here, sandwiched between inputs and outputs.